### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ig1na**
Add waiting animation with Listr and apply for Prettier with its test

### CODE REVIEW

**Comment by ig1na** [comment]
Only -s and -l flags are working for the moment, I added a timeout to the node stack to show the Listr spinner behavior

File: src/rules/prettier/apply.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const cp = require('child_process');
+jest.mock('child_process');
+
+test('Should run npm i prettier -DE command', () => {
+  jest.mock(rootPath + 'package.json', () => ({}), { virtual: true });
+
+  const prettier = new Prettier(rootPath);
+
+  util.promisify.mockImplementation((exec: (cmd: string) => {}) => {
```
Comment (EmmanuelDemey) [inline src/rules/prettier/apply.test.ts]: Why you did not use the same structure as the test for the linter rule I have updated few days ago ?

File: src/stacks/typescript/index.ts [hunk 2]
```diff
@@ -7,7 +7,10 @@ export class TypeScript {
   async isAvailable() {
     try {
       const packageJSON = require(`${this.rootPath}package.json`);
-      return Object.keys(packageJSON.devDependencies).includes('typescript');
+      if (packageJSON.devDependencies !== undefined) {
```
Comment (EmmanuelDemey) [inline src/stacks/typescript/index.ts]: We should maybe create an utility method in order to check if the project as something as a dep or devDep

File: src/stacks/vue-js/index.ts [hunk 3]
```diff
@@ -7,7 +7,10 @@ export class VueJS {
   async isAvailable(): Promise<boolean> {
     try {
       const packageJson = require(`${this.rootPath}/package.json`);
-      return Object.keys(packageJson.dependencies).includes('vue');
+      if (packageJson.dependencies !== undefined) {
```
Comment (EmmanuelDemey) [inline src/stacks/vue-js/index.ts]: We should maybe create an utility method in order to check if the project as something as a dep or devDep

File: src/stacks/node/index.ts [hunk 4]
```diff
@@ -10,7 +10,12 @@ export class Node {
   }
 
   isAvailable(): Promise<boolean> {
-    return existsPaths(`${this.rootPath}package.json`);
+    return new Promise((resolve, reject) =>
```
Comment (EmmanuelDemey) [inline src/stacks/node/index.ts]: Do not forget to remove that

File: src/stacks/github/index.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+@StackRegister.register
+export class GitHub {
+  constructor(readonly rootPath: string = './') {}
+
+  async isAvailable(): Promise<boolean> {
+    const exec = util.promisify(cp.exec);
+
+    return exec(`git -C ${this.rootPath} remote -v`)
+      .then((value: { stdout: string; stderr: string }) => {
```
Comment (EmmanuelDemey) [inline src/stacks/github/index.ts]: Do you need stderr ?
```suggestion
      .then(({ stdout: string }) => {
```

File: src/stacks/angular/index.ts [hunk 6]
```diff
@@ -4,19 +4,13 @@ import { StackRegister } from '../stack-register';
...
-  //   } catch (e) {
-  //     return false;
-  //   }
-  // }
-
   async isAvailable(): Promise<boolean> {
     try {
       const packageJSON = require(`${this.rootPath}package.json`);
-      return Object.keys(packageJSON.dependencies).includes('@angular/core');
+      if (packageJSON.dependencies !== undefined) {
```
Comment (EmmanuelDemey) [inline src/stacks/angular/index.ts]: We should maybe create an utility method in order to check if the project as something as a dep or devDep

File: src/index.ts [hunk 7]
```diff
@@ -7,6 +7,8 @@ import Rule from './rules/rule';
 import { init } from './init/index';
 import { StackRegister } from './stacks/stack-register';
 import * as PromiseBlu from 'bluebird';
+import { ListStacks } from './stacks/list-stacks/index';
+import * as Listr from 'listr';
```
Comment (EmmanuelDemey) [inline src/index.ts]: Is it still useful?

File: package.json [hunk 8]
```diff
@@ -14,27 +14,31 @@
...
-    "bluebird": "^3.5.5",
+    "@types/listr": "0.14.0",
+    "@types/mock-fs": "3.6.30",
+    "@types/winston": "2.4.4",
+    "axios": "0.19.0",
+    "bluebird": "3.5.5",
     "check-node-version": "3.3.0",
     "fs-extra": "8.0.1",
     "inquirer": "6.2.2",
+    "listr": "0.14.3",
```
Comment (EmmanuelDemey) [inline package.json]: Is it still useful?

File: src/rules/linter/index.ts [hunk 9]
```diff
@@ -31,81 +33,93 @@ export class Linter {
   }
 
   private async init(): Promise<void> {
-    // return new Promise(resolve => {
     if (!this.initialized) {
-      return new TypeScript(this.rootPath)
-        .isAvailable()
-        .then(result => {
+      return ListStacks.getStackByName('TypeScript').then(stack => {
```
Comment (EmmanuelDemey) [inline src/rules/linter/index.ts]: We can maybe use the name of stack, and avoid hard coded strings

File: src/rules/vscode-extensions/index.ts [hunk 10]
```diff
@@ -92,9 +90,9 @@ export class VSCodeExtensions {
       };
     }
 
-    answers.forEach(mr => {
-      if (!this.parsedExtensionsFile.recommendations.includes(mr)) {
-        this.parsedExtensionsFile.recommendations.push(mr);
+    answers.forEach(recommendation => {
```
Comment (EmmanuelDemey) [inline src/rules/vscode-extensions/index.ts]: Maybe we can change these three lines with the .filter

**Pull request merged**